src/Uno.UI.Runtime.Skia.Gtk/System/LauncherExtension.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Uno.Extensions.System;
+using Windows.System;
+
+namespace Uno.UI.Runtime.Skia.GTK.Extensions.System
+{
+	internal class LauncherExtension : ILauncherExtension
+	{
+		public LauncherExtension(object owner)
+		{
+		}
+
+		public Task<bool> LaunchUriAsync(Uri uri)
+		{
+			var processStartInfo = new ProcessStartInfo(uri.OriginalString)
+			{
+				UseShellExecute = true,
+				Verb = "open"
+			};
+
+			var process = new Process()
+			{
+				StartInfo = processStartInfo
+			};
+
+			return Task.FromResult(process.Start());
+		}
+
+		public Task<LaunchQuerySupportStatus> QueryUriSupportAsync(Uri uri, LaunchQuerySupportType launchQuerySupportType)
+		{
+			return Task.Run(() =>
+			{
+				var canOpenUri = CheckXdgSettings(uri);
+				return canOpenUri ?
+					LaunchQuerySupportStatus.Available : LaunchQuerySupportStatus.NotSupported;
+			});
+		}
+
+		private bool CheckXdgSettings(Uri uri)
+		{
+			var process = new Process()
+			{
+				StartInfo = new ProcessStartInfo()
+				{
+					FileName = "xdg-settings",
```

Comment:
Is this available on all OSes ?

Reply:
It's part of `xdg-utils`, which should be available for all Linux distros with a desktop-environment.

Reply:
I might add a fallback by [reading the `mimeapps.list` file](https://github.com/trungnt2910/URIScheme/blob/master/URIScheme/Tools/MimeAppsList.cs), where URI schemes should be listed. This method should be available for most Linux Gtk desktops.

Reply:
Tested latest nuget package from artifacts on a Linux machine without `xdg-settings`, and the latest build works.